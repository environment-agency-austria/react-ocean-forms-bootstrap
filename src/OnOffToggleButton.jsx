/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup, Button, Input } from 'reactstrap';
import { fieldMetaShape, fieldShape, withForm } from 'react-ocean-forms';

import FieldLine from './FieldLine';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
class OnOffToggleButton extends React.Component {
  constructor(props) {
    super(props);

    this.onRadioBtnClick = this.onRadioBtnClick.bind(this);
  }

  onRadioBtnClick(value) {
    const { field: { onChange, name } } = this.props;
    onChange({
      target: {
        name,
        value,
      },
    });
  }

  render() {
    const {
      field,
      onLabel,
      offLabel,
      meta: {
        plaintext,
        stringFormatter,
      },
      field: {
        value,
      },
    } = this.props;
    const onValue = stringFormatter(onLabel);
    const offValue = stringFormatter(offLabel);
    const isOn = !(value === '' || value === false);

    if (plaintext) {
      return (
        <FieldLine {...this.props}>
          <Input {...field} plaintext>
            {isOn ? onValue : offValue}
          </Input>
        </FieldLine>
      );
    }

    return (
      <FieldLine {...this.props}>
        <ButtonGroup id={field.id}>
          <Button
            id={`${field.id}-on`}
            color="primary"
            onClick={() => this.onRadioBtnClick(true)}
            active={isOn}
          >
            {onValue}
          </Button>
          <Button
            id={`${field.id}-off`}
            color="primary"
            onClick={() => this.onRadioBtnClick(false)}
            active={!isOn}
          >
            {offValue}
          </Button>
        </ButtonGroup>
      </FieldLine>
    );
  }
}

OnOffToggleButton.displayName = 'OnOffToggleButton';

OnOffToggleButton.defaultProps = {
  onLabel: 'ojs_togglebutton_on',
  offLabel: 'ojs_togglebutton_off',
};

OnOffToggleButton.propTypes = {
  label: PropTypes.string.isRequired,
  onLabel: PropTypes.string,
  offLabel: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
};

export const BaseOnOffToggleButton = OnOffToggleButton;
export default withForm(OnOffToggleButton);
