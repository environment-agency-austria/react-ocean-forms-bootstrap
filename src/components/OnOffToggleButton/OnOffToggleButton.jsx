/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup, Button, Input } from 'reactstrap';
import {
  FormText,
  fieldMetaShape,
  fieldShape,
  withForm,
} from 'react-ocean-forms';

import FieldLine from '../FieldLine';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
class OnOffToggleButton extends React.Component {
  constructor(props) {
    super(props);

    this.onRadioBtnClick = this.onRadioBtnClick.bind(this);
  }

  onRadioBtnClick(value) {
    const { field: { onChange, name } } = this.props;
    onChange({
      target: {
        name,
        value,
      },
    });
  }

  render() {
    const {
      field,
      onLabel,
      offLabel,
      meta: {
        plaintext,
      },
      field: {
        value,
      },
    } = this.props;

    const isOn = !(value === '' || value === false);

    if (plaintext) {
      return (
        <FieldLine {...this.props}>
          <Input {...field} plaintext>
            <FormText text={isOn ? onLabel : offLabel} />
          </Input>
        </FieldLine>
      );
    }

    return (
      <FieldLine {...this.props}>
        <ButtonGroup id={field.id}>
          <Button
            id={`${field.id}-on`}
            color="primary"
            onClick={() => this.onRadioBtnClick(true)}
            outline
            active={isOn}
            disabled={field.disabled}
          >
            <FormText text={onLabel} />
          </Button>
          <Button
            id={`${field.id}-off`}
            color="primary"
            onClick={() => this.onRadioBtnClick(false)}
            outline
            active={!isOn}
            disabled={field.disabled}
          >
            <FormText text={offLabel} />
          </Button>
        </ButtonGroup>
      </FieldLine>
    );
  }
}

OnOffToggleButton.displayName = 'OnOffToggleButton';

OnOffToggleButton.defaultProps = {
  onLabel: 'ojs_togglebutton_on',
  offLabel: 'ojs_togglebutton_off',
};

OnOffToggleButton.propTypes = {
  label: PropTypes.string.isRequired,
  onLabel: PropTypes.string,
  offLabel: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  field: fieldShape.isRequired,
};

export const BaseOnOffToggleButton = OnOffToggleButton;
export default withForm(OnOffToggleButton);
