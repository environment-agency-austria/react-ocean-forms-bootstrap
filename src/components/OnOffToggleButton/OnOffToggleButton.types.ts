import { IUseFieldProps } from 'react-ocean-forms';

import { IBaseFieldLineProps } from '../FieldLine';

/**
 * Props for the OnOffToggleButton component
 */
export interface IOnOffToggleButtonProps<TSubmitValue> extends IUseFieldProps<boolean, TSubmitValue>, IBaseFieldLineProps {
  /**
   * Label for the on-button
   */
  onLabel?: string;
  /**
   * Label for the off-button
   */
  offLabel?: string;
}
