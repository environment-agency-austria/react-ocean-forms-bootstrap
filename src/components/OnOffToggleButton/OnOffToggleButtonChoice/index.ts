export * from './OnOffToggleButtonChoice';
export * from './OnOffToggleButtonChoice.types';
