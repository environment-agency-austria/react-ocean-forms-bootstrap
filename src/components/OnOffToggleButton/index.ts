export * from './OnOffToggleButton';
export * from './OnOffToggleButton.types';
