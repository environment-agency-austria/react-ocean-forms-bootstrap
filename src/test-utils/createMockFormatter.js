export default () => jest.fn().mockImplementation(id => id);
