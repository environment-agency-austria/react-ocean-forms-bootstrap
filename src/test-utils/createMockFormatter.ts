export const createMockFormatter = (): jest.Mock => jest.fn().mockImplementation((id: string) => id);
