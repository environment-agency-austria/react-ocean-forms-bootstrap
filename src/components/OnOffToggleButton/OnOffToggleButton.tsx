/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { useCallback } from 'react';

import { useField } from 'react-ocean-forms';
import {  ButtonGroup } from 'reactstrap';

import { FieldLine } from '../FieldLine';
import { IOnOffToggleButtonProps } from './OnOffToggleButton.types';
import { PlaintextOnOffToggleButton } from './PlaintextOnOffToggleButton';
import { OnOffToggleButtonChoice } from './OnOffToggleButtonChoice';

/**
 * Component for displaying bootstrap
 * form groups with an html input and
 * oForm support
 */
export const OnOffToggleButton = <TSubmitValue extends unknown = boolean>(props: IOnOffToggleButtonProps<TSubmitValue>): JSX.Element => {
  const {
    onLabel = 'ojs_togglebutton_on',
    offLabel = 'ojs_togglebutton_off',
    ...rest
  } = props;

  const { fieldProps, metaProps } = useField(rest);

  const { onChange } = fieldProps;
  const onRadioBtnClick = useCallback((value: boolean) => {
    onChange({
      target: {
        value,
      }
    });
  }, [onChange]);

  if (metaProps.plaintext) {
    return (
      <PlaintextOnOffToggleButton
        fieldProps={fieldProps}
        metaProps={metaProps}
        onLabel={onLabel}
        offLabel={offLabel}
        {...rest}
      />
    );
  }

  return (
    <FieldLine {...rest} fieldProps={fieldProps} metaProps={metaProps}>
      <ButtonGroup id={fieldProps.id}>
        <OnOffToggleButtonChoice
          variant="on"
          label={onLabel}
          fieldProps={fieldProps}
          onClick={onRadioBtnClick}
        />
        <OnOffToggleButtonChoice
          variant="off"
          label={offLabel}
          fieldProps={fieldProps}
          onClick={onRadioBtnClick}
        />
      </ButtonGroup>
    </FieldLine>
  );
};
