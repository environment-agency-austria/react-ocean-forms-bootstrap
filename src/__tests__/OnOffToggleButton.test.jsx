import React from 'react';
import { shallow } from 'enzyme';
import createMockFormatter from '../test-utils/createMockFormatter';

import { BaseOnOffToggleButton } from '../OnOffToggleButton';

describe('<BaseOnOffToggleButton />', () => {
  const FIELD_ID = 'field0';
  const FIELD_NAME = 'field0';
  const FIELD_LABEL = 'field0';

  const meta = {
    valid: true,
    error: undefined,
    isValidating: undefined,
    stringFormatter: createMockFormatter(),
    plaintext: false,
  };

  const field = {
    value: '',
    invalid: false,
    id: FIELD_ID,
    name: FIELD_NAME,
    disabled: false,
    onChange: jest.fn(),
    onBlur: jest.fn(),
  };

  const setup = props => shallow((
    <BaseOnOffToggleButton
      label={FIELD_LABEL}
      meta={meta}
      field={field}
      onLabel="on"
      offLabel="off"
      {...props}
    />
  ));

  it('should render without crashing', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render plaintext Button', () => {
    meta.plaintext = true;
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  const testButton = (value) => {
    describe(`${value} Button`, () => {
      const wrapper = setup();
      const buttonOn = wrapper.find(`#field0-${value}`).dive();

      it('should render labels for button', () => {
        expect(buttonOn.text()).toBe(value);
      });

      it('should trigger onChange on click', () => {
        buttonOn.simulate('click');
        expect(field.onChange).toHaveBeenCalledWith({ target: { name: FIELD_NAME, value: value === 'on' } });
      });
    });
  };

  testButton('on');

  testButton('off');
});
