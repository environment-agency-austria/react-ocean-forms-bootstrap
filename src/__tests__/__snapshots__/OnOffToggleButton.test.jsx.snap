// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseOnOffToggleButton /> should render plaintext Button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": false,
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "",
      }
    }
    label="field0"
    meta={
      Object {
        "error": undefined,
        "isValidating": undefined,
        "plaintext": true,
        "stringFormatter": [MockFunction] {
          "calls": Array [
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
          ],
        },
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Input
        disabled={false}
        id="field0"
        invalid={false}
        name="field0"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        plaintext={true}
        type="text"
        value=""
      >
        off
      </Input>,
      "field": Object {
        "disabled": false,
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "",
      },
      "info": undefined,
      "label": "field0",
      "meta": Object {
        "error": undefined,
        "isValidating": undefined,
        "plaintext": true,
        "stringFormatter": [MockFunction] {
          "calls": Array [
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
          ],
        },
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
      "prefix": undefined,
      "suffix": undefined,
      "validators": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "off",
        "disabled": false,
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "plaintext": true,
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": "off",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Input
          disabled={false}
          id="field0"
          invalid={false}
          name="field0"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          plaintext={true}
          type="text"
          value=""
        >
          off
        </Input>,
        "field": Object {
          "disabled": false,
          "id": "field0",
          "invalid": false,
          "name": "field0",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "",
        },
        "info": undefined,
        "label": "field0",
        "meta": Object {
          "error": undefined,
          "isValidating": undefined,
          "plaintext": true,
          "stringFormatter": [MockFunction] {
            "calls": Array [
              Array [
                "on",
              ],
              Array [
                "off",
              ],
              Array [
                "on",
              ],
              Array [
                "off",
              ],
              Array [
                "on",
              ],
              Array [
                "off",
              ],
              Array [
                "on",
              ],
              Array [
                "off",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
            ],
          },
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
        "prefix": undefined,
        "suffix": undefined,
        "validators": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "off",
          "disabled": false,
          "id": "field0",
          "invalid": false,
          "name": "field0",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "plaintext": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": "off",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<BaseOnOffToggleButton /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnOffToggleButton
    field={
      Object {
        "disabled": false,
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "",
      }
    }
    label="field0"
    meta={
      Object {
        "error": undefined,
        "isValidating": undefined,
        "plaintext": false,
        "stringFormatter": [MockFunction] {
          "calls": Array [
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
          ],
        },
        "valid": true,
      }
    }
    offLabel="off"
    onLabel="on"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ButtonGroup
        id="field0"
        role="group"
        tag="div"
      >
        <Button
          active={false}
          color="primary"
          id="field0-on"
          onClick={[Function]}
          tag="button"
        >
          on
        </Button>
        <Button
          active={true}
          color="primary"
          id="field0-off"
          onClick={[Function]}
          tag="button"
        >
          off
        </Button>
      </ButtonGroup>,
      "field": Object {
        "disabled": false,
        "id": "field0",
        "invalid": false,
        "name": "field0",
        "onBlur": [MockFunction],
        "onChange": [MockFunction],
        "value": "",
      },
      "info": undefined,
      "label": "field0",
      "meta": Object {
        "error": undefined,
        "isValidating": undefined,
        "plaintext": false,
        "stringFormatter": [MockFunction] {
          "calls": Array [
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
            Array [
              "on",
            ],
            Array [
              "off",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
            Object {
              "isThrow": false,
              "value": "on",
            },
            Object {
              "isThrow": false,
              "value": "off",
            },
          ],
        },
        "valid": true,
      },
      "offLabel": "off",
      "onLabel": "on",
      "prefix": undefined,
      "suffix": undefined,
      "validators": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            color="primary"
            id="field0-on"
            onClick={[Function]}
            tag="button"
          >
            on
          </Button>,
          <Button
            active={true}
            color="primary"
            id="field0-off"
            onClick={[Function]}
            tag="button"
          >
            off
          </Button>,
        ],
        "id": "field0",
        "role": "group",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": "on",
            "color": "primary",
            "id": "field0-on",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "on",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": "off",
            "color": "primary",
            "id": "field0-off",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "off",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ButtonGroup
          id="field0"
          role="group"
          tag="div"
        >
          <Button
            active={false}
            color="primary"
            id="field0-on"
            onClick={[Function]}
            tag="button"
          >
            on
          </Button>
          <Button
            active={true}
            color="primary"
            id="field0-off"
            onClick={[Function]}
            tag="button"
          >
            off
          </Button>
        </ButtonGroup>,
        "field": Object {
          "disabled": false,
          "id": "field0",
          "invalid": false,
          "name": "field0",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "",
        },
        "info": undefined,
        "label": "field0",
        "meta": Object {
          "error": undefined,
          "isValidating": undefined,
          "plaintext": false,
          "stringFormatter": [MockFunction] {
            "calls": Array [
              Array [
                "on",
              ],
              Array [
                "off",
              ],
              Array [
                "on",
              ],
              Array [
                "off",
              ],
              Array [
                "on",
              ],
              Array [
                "off",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
              Object {
                "isThrow": false,
                "value": "on",
              },
              Object {
                "isThrow": false,
                "value": "off",
              },
            ],
          },
          "valid": true,
        },
        "offLabel": "off",
        "onLabel": "on",
        "prefix": undefined,
        "suffix": undefined,
        "validators": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Button
              active={false}
              color="primary"
              id="field0-on"
              onClick={[Function]}
              tag="button"
            >
              on
            </Button>,
            <Button
              active={true}
              color="primary"
              id="field0-off"
              onClick={[Function]}
              tag="button"
            >
              off
            </Button>,
          ],
          "id": "field0",
          "role": "group",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": "on",
              "color": "primary",
              "id": "field0-on",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": "on",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "children": "off",
              "color": "primary",
              "id": "field0-off",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": "off",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
